sympy/solvers/tests/test_solveset.py

```diff
@@ -1181,6 +1181,16 @@ def test_solveset():
     assert solveset(atan(log(x)) > 0, x, domain=Interval.open(0, oo)) == Interval.open(1, oo)
 
 
+@_both_exp_pow
+def test_double_exp():
+    assert dumeq(solveset(exp(exp(x))-5, x),\
+         imageset(Lambda(((k, n),), 2*k*I*pi + log(Abs(2*n*I*pi + log(5)))), ProductSet(S.Integers, S.Integers)))
+
+    assert dumeq(solveset(d*exp(exp(a*x + b)) + c, x),\
+         imageset(Lambda(((k, n),), (2*k*I*pi - b + log(Abs(I*(2*n*pi + arg(-c/d)) + log(Abs(c/d)))))/a),\
```

Comment:
I think this `arg(-c/d)` is not correct and also there should be 2 `args` for 2 `logs` as in L 353 - 356 in solveset.py...(for one `log`)
```
elif isinstance(g_ys, FiniteSet):
            exp_invs = Union(*[imageset(Lambda(n, I*(2*n*pi + arg(g_y)) +
                                               log(Abs(g_y))), S.Integers)
                               for g_y in g_ys if g_y != 0])
```